From 6eba239a5bb553fd557b7d78f7da8f0059618b9e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B8rgen=20Ibsen?= <jibsen@users.noreply.github.com>
Date: Thu, 13 Sep 2018 19:31:23 +0200
Subject: [PATCH 06/30] Fix auto detect of bundled mode (#704)

Set bundled mode to ON when parent directory is not empty. Due to the
peculiarities of CMake if, comparing an undefined variable to the empty
string is false, so this likely never triggered.
---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99b9258..6f1c5a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,13 +11,13 @@ project(brotli C)
 # we'll use the BROTLI_BUNDLED_MODE variable to let them do that; just
 # set it to OFF in your project before you add_subdirectory(brotli).
 get_directory_property(BROTLI_PARENT_DIRECTORY PARENT_DIRECTORY)
-if(BROTLI_BUNDLED_MODE STREQUAL "")
+if(NOT DEFINED BROTLI_BUNDLED_MODE)
   # Bundled mode hasn't been set one way or the other, set the default
   # depending on whether or not we are the top-level project.
   if(BROTLI_PARENT_DIRECTORY)
-    set(BROTLI_BUNDLED_MODE OFF)
-  else()
     set(BROTLI_BUNDLED_MODE ON)
+  else()
+    set(BROTLI_BUNDLED_MODE OFF)
   endif()
 endif()
 mark_as_advanced(BROTLI_BUNDLED_MODE)
-- 
2.19.1

